### PR TITLE
Improve error message when Node.js is unresponsive

### LINKED ISSUE #1847: Rephrase timeout error to be more user friendly

### PR DESCRIPTION
**Pull request by saberduck**
Fixes #1847

### CODE REVIEW

File: sonar-javascript-plugin/src/main/resources/static/documentation.md [hunk 1]
```diff
@@ -28,6 +28,19 @@ There are 2 built-in rule profiles for JavaScript: `Sonar way` (default) and `So
 * `Sonar way` profile is activated by default. It defines a trimmed list of high-value/low-noise rules useful in almost any JS development context.
 * `Sonar way Recommended` contains all rules from `Sonar way`, plus more rules that mandate high code readability and long-term project evolution.
 
+## Troubleshooting
+
+### Slow or unresponsive analysis
+
+On a big project it can happen that analyzer needs to allocate a lot of memory to analyze the project. This would manifest by analysis getting stuck and following stacktrace might appear in the logs
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/resources/static/documentation.md]: should not it be `This would be manifested` in passive?

Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/resources/static/documentation.md]: `a lot` -> `more`

Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/resources/static/documentation.md]: **the** following

File: sonar-javascript-plugin/src/main/resources/static/documentation.md [hunk 2]
```diff
@@ -28,6 +28,19 @@ There are 2 built-in rule profiles for JavaScript: `Sonar way` (default) and `So
...
+
+```
+ERROR: Failed to get response while analyzing [file].ts
+java.io.InterruptedIOException: timeout
+```   
+You can use `sonar.javascript.node.maxspace` property to allow the analyzer to use more memory. Set this property to `4096` or `8192` for big projects. This property should be set in `sonar-project.properties` file or on command line for scanner (with `-Dsonar.javascript.node.maxspace=4096`).
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/resources/static/documentation.md]: Are you sure that the only 2 ways to specify properties? Should not we use some more generic phasing?

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImpl.java [hunk 3]
```diff
@@ -192,6 +193,13 @@ private String request(String json, String endpoint) throws IOException {
     try (Response response = client.newCall(request).execute()) {
       // in this case response.body() is never null (according to docs)
       return response.body().string();
+    } catch (InterruptedIOException e) {
+      if ("timeout".equals(e.getMessage())) {
+        String msg = "eslint-bridge Node.js process is unresponsive. This is most likely caused by process running out of memory." +
+          " Consider setting sonar.javascript.node.maxspace to higher value (e.g. 4096).";
+        LOG.error(msg);
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImpl.java]: can we test this log?

File: sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImplTest.java [hunk 4]
```diff
@@ -154,10 +154,9 @@ public void should_throw_if_failed_to_start() throws Exception {
     eslintBridgeServer = createEslintBridgeServer("throw.js");
     eslintBridgeServer.deploy();
 
-    thrown.expect(NodeCommandException.class);
-    thrown.expectMessage("Failed to start server (" + TEST_TIMEOUT_SECONDS + "s timeout)");
-
-    eslintBridgeServer.startServer(context);
+    assertThatThrownBy(() -> eslintBridgeServer.startServer(context))
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImplTest.java]: is it just refactoring?

File: sonar-javascript-plugin/src/main/resources/static/documentation.md (hunk 2 above)
Comment (saberduck) [reply sonar-javascript-plugin/src/main/resources/static/documentation.md]: I prefer to be concrete, if we are generic people will not understand and set the property on the server side. This should work with all scanners afaik

File: sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImplTest.java (hunk 4 above)
Comment (saberduck) [reply sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImplTest.java]: no, for some reason this assertion doesn't work on windows anymore. Assertj is a bit better

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImpl.java (hunk 3 above)
Comment (saberduck) [reply sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImpl.java]: added test

**Pull request merged**